internal/messaging/response_handler.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2022-2023, Chain4Travel AG. All rights reserved.
+ * See the file LICENSE for licensing terms.
+ */
+
+package messaging
+
+import (
+	typesv1alpha "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/types/v1alpha"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/chain4travel/camino-messenger-bot/internal/tvm"
+	"github.com/chain4travel/caminotravelvm/actions"
+	"github.com/chain4travel/caminotravelvm/consts"
+	"go.uber.org/zap"
+	"strconv"
+)
+
+var _ ResponseHandler = (*TvmResponseHandler)(nil)
+
+type ResponseHandler interface {
+	HandleResponse(ctx context.Context, msgType MessageType, request *RequestContent, response *ResponseContent)
+}
+type TvmResponseHandler struct {
+	tvmClient *tvm.Client
+	logger    *zap.SugaredLogger
+}
+
+func (h *TvmResponseHandler) HandleResponse(ctx context.Context, msgType MessageType, request *RequestContent, response *ResponseContent) {
+	switch msgType {
+	case MintRequest: // distributor will post-process a mint request to buy the returned NFT
+		if response.MintResponse.Header == nil {
+			response.MintResponse.Header = &typesv1alpha.ResponseHeader{}
+		}
+		if response.MintTransactionId == "" {
+			addErrorToResponseHeader(response, "missing mint transaction id")
+			return
+		}
+		mintID, err := ids.FromString(response.MintTransactionId)
+		if err != nil {
+			addErrorToResponseHeader(response, fmt.Sprintf("error parsing mint transaction id: %v", err))
+			return
+		}
+
+		success, txID, err := h.tvmClient.SendTxAndWait(ctx, transferNFTAction(h.tvmClient.Address(), mintID))
```

Comment:
As far as I understand, `TransferNFTAction` also contains the "buy operation" logic. I also checked the `TransferNFT` logic in travelvm and it seems to like _in every transfer_ the price will be subtracted from the actor.